src/main/User/Services/DeleteUserService.java

```diff
@@ -0,0 +1,118 @@
+package User.Services;
+
+import Config.Message;
+import Config.Service;
+import Database.User.UserDao;
+import PDF.PDFType;
+import Security.SecurityUtils;
+import User.User;
+import User.UserMessage;
+import Validation.ValidationUtils;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import com.mongodb.client.gridfs.model.GridFSFile;
+import com.mongodb.client.model.Filters;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+public class DeleteUserService implements Service {
+  private MongoDatabase db;
+  private UserDao userDao;
+  private String username;
+  private String password;
+  private User user;
+
+  public DeleteUserService(MongoDatabase db, UserDao userDao, String username, String password) {
+    this.db = db;
+    this.userDao = userDao;
+    this.username = username;
+    this.password = password;
+  }
+
+  @Override
+  public Message executeAndGetResponse() {
+    // validation
+    if (!ValidationUtils.isValidPassword(password)) {
+      log.info("Invalid password");
+      return UserMessage.AUTH_FAILURE;
+    }
+
+    // get user
+    Optional<User> optionalUser = userDao.get(username);
+    if (optionalUser.isEmpty()) {
+      log.info(username + " not found");
+      return UserMessage.USER_NOT_FOUND;
+    }
+    user = optionalUser.get();
+
+    // verify password
+    if (!verifyPassword(password, user.getPassword())) {
+      log.info("Incorrect password");
+      return UserMessage.AUTH_FAILURE;
+    }
+
+    deleteUserFiles();
+    userDao.delete(user.getUsername());
+    return UserMessage.SUCCESS;
+  }
+
+  public void deleteUserFiles() {
+    // Form
+    GridFSBucket gridBucket = GridFSBuckets.create(db, PDFType.FORM.toString());
```

Comment:
You can add a new function to the PdfDao called deleteAll() that you could move all the logic from here to there